src/pages/_middleware.ts

```diff
@@ -3,6 +3,36 @@ import {ACCESS_TOKEN_KEY} from '../config'
 
 const PUBLIC_FILE = /\.(.*)$/
 
+const CIO_COOKIE_KEY = 'cio_id'
+const CIO_BASE_URL = `https://beta-api.customer.io/v1/api/`
+
+const loadCio = async (cioId: string, customer?: any) => {
+  try {
+    if (customer) {
```

Comment:
if it's already in the cookie, we can just use it here instead of fetching from CIO and save some requests/speed